### PR TITLE
[ready-for-core-team-review]CRM-16189, fixed code to check if financial type id is present in $pa…

### PR DESCRIPTION
**Pull request by pradpnayak**
…rams

---
- CRM-16189: Improve support for Accrual Method bookkeeping
  https://issues.civicrm.org/jira/browse/CRM-16189

### CODE REVIEW

**Comment by pradpnayak** [comment]
Is test failure related to PR?

**Comment by pradpnayak** [comment]
jenkins, retest this please

File: CRM/Event/BAO/Event.php [hunk 1]
```diff
@@ -96,13 +96,12 @@ public static function add(&$params) {
...
 
     // CRM-16189
-    $isError = CRM_Financial_BAO_FinancialAccount::validateFinancialType(
-      CRM_Utils_Array::value('financial_type_id', $params)
-    );
-    if ($isError) {
-      throw new CRM_Core_Exception(ts('Deferred revenue account is not configured for selected financial type. Please have an administrator set up the deferred revenue account at Administer > CiviContribute > Financial Accounts, then configure it for financial types at Administer > CiviContribution > Financial Types, Accounts'));
+    if (!empty($params['financial_type_id'])) {
+      $isError = CRM_Financial_BAO_FinancialAccount::validateFinancialType($params['financial_type_id']);
+      if ($isError) {
```
Comment (JoeMurray) [inline CRM/Event/BAO/Event.php]: switch to catching exception here

**Comment by pradpnayak** [comment]
Added the changes

File: CRM/Event/BAO/Event.php [hunk 2]
```diff
@@ -96,13 +96,12 @@ public static function add(&$params) {
...
+        throw new CRM_Core_Exception(ts('Deferred revenue account is not configured for selected financial type. Please have an administrator set up the deferred revenue account at Administer > CiviContribute > Financial Accounts, then configure it for financial types at Administer > CiviContribution > Financial Types, Accounts'));
+      }
```
Comment (JoeMurray) [inline CRM/Event/BAO/Event.php]: Shouldn't you be catching the exception thrown in CRM_Financial_BAO_FinancialAccount::validateFinancialType() and using its error message?

**Comment by JoeMurray** [comment]
jenkins, please test this

**Comment by JoeMurray** [comment]
jenkins, please retest this

**Comment by JoeMurray** [comment]
jenkins, retest this please

**Pull request merged**